include/level_object_entries.h

```diff
@@ -0,0 +1,373 @@
+#ifndef _LEVEL_OBJECT_ENTRIES_H_
+#define _LEVEL_OBJECT_ENTRIES_H_
+
+    /* Size: 8 bytes. Shared across every entry. */
+typedef struct LevelObjectEntryCommon {
```

Comment:
Maybe you want to add something in the style guide about this, but the first declaration of the struct name _usually_ isn't necessary. e.g. this could just be
```
typedef struct {
    u8 object_id;
    u8 size;
    s16 x, y, z;
} LevelObjectEntryCommon;
```
Same with all the other structs below.

Reply:
It isn't required, but it is a personal preference thing. I've always liked having the name above rather than below. However I don't want to write `struct` every time I declare the struct, so the bottom name is required in that sense. All of the structs in `structs.h`should also follow this style if you look in that file.

Reply:
That SGTM. My default is to omit them, and I'm sure I've committed some structs like that in the past. Looking through the codebase that seems to be the case in other areas as well. I think clarifying this in the style guide would be good.